src/core/stream.c

```diff
@@ -502,6 +502,7 @@ QuicStreamTryCompleteShutdown(
     )
 {
     if (!Stream->Flags.ShutdownComplete &&
+        !Stream->Flags.ReceiveDataPending &&
```

Comment:
Does this solve both cases in #1384 ? Just want to make sure this doesn't just fix 1 of the 2 cases.

Reply:
Yes, it should. This flag is set so long as there is any unconsumed (by the app) data buffered. Data is not considered "consumed" until the receive has completed and the app has indicated the correct length/offset.